### PR TITLE
cmake: Always take executable for build machine

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

**Comment by xclaesse** [comment]
When cross compiling a meson project with a cmake subproject, it fails to find cmake executable for the host machine, but it should not be needed. As far as I know everything about for_machine in CMakeExecutor can be removed, no?

**Comment by mensinda** [comment]
Not sure. I never tried to cross-compiling. You know probably more than I do :)

**Comment by xclaesse** [comment]
> I never tried to cross-compiling.

Spoiler alert: it doesn't work :D It's also missing all c_args and friends from the cross file.

**Comment by ignatenkobrain** [comment]
Flake8 detected 7 issues on 484a79035686e94f604a371dc23286d4340c570f
Visit https://sider.review/gh/repos/19784232/pulls/5961 to review the issues.

<sub>posted by Sider</sub>

File: mesonbuild/cmake/executor.py [hunk 1]
```diff
@@ -15,8 +15,9 @@
 # This class contains the basic functionality needed to run any interpreter
 # or an interpreter-based tool.
 
+from .common import CMakeException
 from .. import mlog, mesonlib
-from ..mesonlib import PerMachine, Popen_safe, version_compare, MachineChoice
+from ..mesonlib import Popen_safe, version_compare, MachineChoice
```
Comment (ignatenkobrain) [inline mesonbuild/cmake/executor.py]: __[Flake8]__ [F401] '..mesonlib.version_compare' imported but unused ([link](https://sider.review/gh/repos/19784232/pulls/5961#issue-4619617))

<sub>posted by Sider</sub>

**Comment by mensinda** [comment]
I am going to close this in favor of #7816 (also, there are a lot of conflicts in this PR anyway). Also, the transparent dependency fallback for CMake subprojects won't be happening any time soon because of design reasons anyway.

**Pull request closed without merging**